pandas/core/arrays/_mixins.py

```diff
@@ -240,3 +240,24 @@ def _reduce(self, name: str, skipna: bool = True, **kwargs):
         else:
             msg = f"'{type(self).__name__}' does not implement reduction '{name}'"
             raise TypeError(msg)
+
+    # ------------------------------------------------------------------------
+
+    def __repr__(self) -> str:
```

Comment:
isn't this the same def that's in pandas/core/arrays/base.py?

Reply:
no, this dispatches to that in the 1D case, then handles the 2D case here

Reply:
how about we make methods (name _1d, _2d) that are well named for these (and then it will be much more obvious that this is happening).

Reply:
i guess we could.  i think its pretty clear here since the first two lines are `if self.ndim == 1: return super().__repr__()`

Reply:
updated per request + green